week-2-wide-perspective/w02e02-data-manipulation/src/model/orders.ts

```diff
@@ -0,0 +1,56 @@
+import { BehaviorSubject, map, Observable } from "rxjs";
+import {
+  generateId,
+  getBestSale,
+  getByYear,
+  getTotalIncome,
+  wait,
+} from "../orders/helpers";
+
+export type OrderType = {
+  id: string;
+  orderNumber: string;
+  orderDate: string;
+  sale: number;
+};
+
+const createOrder = (sale: number, orderNumber: string): OrderType => ({
+  id: generateId(),
+  orderDate: new Date().toISOString(),
+  sale,
+  orderNumber,
+});
+
+export class Orders {
+  #source = new BehaviorSubject<OrderType[]>([]);
+
+  constructor(orders: OrderType[]) {
+    this.#source.next(orders);
+  }
+
+  getBestSale(): Observable<number> {
+    return this.#source.pipe(map(getBestSale));
+  }
+
+  getTotalIncome(): Observable<number> {
+    return this.#source.pipe(map(getTotalIncome));
```

Comment:
bardzo ładnie zrealizowany temat `helpers` gdzie pozamykałeś @adamJacewicz te funkcyjki sobie i mają swoje miejsce.

Ja bym jedynie dał inną nazwe na ich mieszkanie `helpers.ts` to dla mnie zbyt "generyczna" nazwa. nie mówiąca zbyt wiele.